core/capabilities/triggers/logevent/trigger.go

```diff
@@ -0,0 +1,178 @@
+package logevent
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/core"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+// Log Event Trigger Capability Request Config Details
+type RequestConfig struct {
+	ContractName         string                     `json:"contractName"`
+	ContractAddress      common.Address             `json:"contractAddress"`
+	ContractEventName    string                     `json:"contractEventName"`
+	ContractReaderConfig evmtypes.ChainReaderConfig `json:"contractReaderConfig"`
+}
+
+// LogEventTrigger struct to listen for Contract events using ContractReader gRPC client
+// in a loop with a periodic delay of pollPeriod milliseconds, which is specified in
+// the job spec
+type logEventTrigger struct {
+	ch   chan<- capabilities.TriggerResponse
+	lggr logger.Logger
+	ctx  context.Context
+
+	// Contract address and Event Signature to monitor for
+	reqConfig      *RequestConfig
+	contractReader types.ContractReader
+	relayer        core.Relayer
+	startBlockNum  uint64
+
+	// Log Event Trigger config with pollPeriod and lookbackBlocks
+	logEventConfig LogEventConfig
+	ticker         *time.Ticker
+	done           chan bool
+}
+
+// Construct for logEventTrigger struct
```

Comment:
nit: vaguely feels like redundant comment.

Reply:
Sure, I'll try to make this more explanatory